spec/lib/tasks/publishing_api_spec.rb

```diff
@@ -0,0 +1,106 @@
+      end
+
+      expect { Rake::Task["publishing_api:publish_finders"].invoke }.to output.to_stdout
+
+      content_ids.each do |content_id|
+        assert_publishing_api_put_content(content_id)
+        assert_publishing_api_patch_links(content_id)
+        assert_publishing_api_publish(content_id)
+      end
+    end
+
+    context "an error triggers the rescue state" do
+      it "returns an error message" do
+        stub_any_publishing_api_put_content.and_raise(GdsApi::HTTPServerError.new(500))
+        error_message = %r{Error publishing finder: #<GdsApi::HTTPServerError: GdsApi::HTTPServerError>}
+
+        expect { Rake::Task["publishing_api:publish_finders"].invoke }.to output(error_message).to_stdout
+      end
+    end
+  end
+
+  describe "publishing_api:publish_finder" do
+    before(:each) do
+      Rake::Task["publishing_api:publish_finder"].reenable
+      stub_any_publishing_api_put_content
+      stub_any_publishing_api_patch_links
+      stub_any_publishing_api_publish
+    end
+
+    context "incorrect file name is given" do
+      it "returns an error message" do
+        error_message = %r{#<RuntimeError: Could not find file: lib/documents/schemas/aaib_reportings.json>}
+
+        expect { Rake::Task["publishing_api:publish_finder"].invoke("aaib_reportings") }.to output(error_message).to_stdout
+      end
+    end
+
+    context "correct file name given" do
+      context "no server error present" do
+        it "publishes a single finder to the Publishing API" do
+          content_id = "b7574bba-969f-4c49-855a-ae1586258ff6"
+
+          expect { Rake::Task["publishing_api:publish_finder"].invoke("aaib_reports") }.to output.to_stdout
+
+          assert_publishing_api_put_content(content_id)
+          assert_publishing_api_patch_links(content_id)
+          assert_publishing_api_publish(content_id)
+        end
+      end
+
+      context "server error present" do
+        it "returns an error message" do
+          stub_any_publishing_api_put_content.and_raise(GdsApi::HTTPServerError.new(500))
+          error_message = %r{Error publishing finder: #<GdsApi::HTTPServerError: GdsApi::HTTPServerError>}
+
+          expect { Rake::Task["publishing_api:publish_finder"].invoke("aaib_reports") }.to output(error_message).to_stdout
+        end
+      end
+    end
+  end
+
+  describe "publishing_api:patch_document_type_links" do
+    let(:cma_cases) { FactoryBot.create_list(:cma_case, 10) }
+
+    before(:each) do
+      Rake::Task["publishing_api:patch_document_type_links"].reenable
+      stub_any_publishing_api_patch_links
+      stub_publishing_api_has_content(cma_cases, hash_including(document_type: CmaCase.document_type))
+    end
+
+    it "patches links for all CMA cases" do
+      message = %r{Links patched for 10 cma_case documents}
+
+      expect {
+        Rake::Task["publishing_api:patch_document_type_links"].invoke("cma_case")
+      }.to output(message).to_stdout
+    end
```

Comment:
I wonder if we should have an expectation here that it actually hits the `patch_links` endpoint for each case rather than only relying on the count

```
cma_cases.each do |cma_case|
  assert_publishing_api_patch_links(cma_case['content_id'])
end
```
Wdyt?